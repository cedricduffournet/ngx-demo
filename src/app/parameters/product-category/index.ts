export * from './parameter-product-category.component';
