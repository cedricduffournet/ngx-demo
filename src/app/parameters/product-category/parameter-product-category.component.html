<app-product-category-list-view></app-product-category-list-view>
