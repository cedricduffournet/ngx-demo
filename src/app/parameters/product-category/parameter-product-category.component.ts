import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-parameter-product-category',
  templateUrl: './parameter-product-category.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ParameterProductCategoryComponent implements OnInit {
  constructor() {}

  ngOnInit() {}
}
