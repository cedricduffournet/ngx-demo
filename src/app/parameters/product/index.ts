export * from './parameter-product.component';
