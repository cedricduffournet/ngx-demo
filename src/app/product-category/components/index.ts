export * from './product-category-add/product-category-add.component';
export * from './product-category-update/product-category-update.component';
export * from './product-category-delete/product-category-delete.component';
export * from './product-category-form/product-category-form.component';
export * from './product-category-item/product-category-item.component';
export * from './product-category-items/product-category-items.component';
