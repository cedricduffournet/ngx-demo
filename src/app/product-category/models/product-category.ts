import { schema } from 'normalizr';

export interface ProductCategory {
  id: number;
  name: string;
}

export const productCategorySchema = new schema.Entity('productCategories');
