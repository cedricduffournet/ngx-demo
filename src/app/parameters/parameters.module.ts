import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { TranslateModule } from '@ngx-translate/core';

import { PageInnerModule } from '@app/shared/page-inner/page-inner.module';
import { CivilityModule } from '@app/civility/civility.module';
import { ProductCategoryModule } from '@app/product-category/product-category.module';
import { ProductModule } from '@app/product/product.module';

import { ListParametersComponent } from '@app/parameters/containers';
import { ReferencedTableMenuComponent } from '@app/parameters/components';
import { ParameterCivilityComponent } from '@app/parameters/civility';
import { ParameterProductCategoryComponent } from '@app/parameters/product-category';
import { ParameterProductComponent, ParameterProductDetailsComponent, ParameterProductAddComponent } from '@app/parameters/product';

import { routing } from './parameters.routing';

@NgModule({
  imports: [
    CommonModule,
    TranslateModule,
    PageInnerModule,
    CivilityModule,
    routing,
    ReactiveFormsModule,
    ProductCategoryModule,
    ProductModule
  ],
  declarations: [
    ListParametersComponent,
    ReferencedTableMenuComponent,
    ParameterCivilityComponent,
    ParameterProductCategoryComponent,
    ParameterProductComponent,
    ParameterProductDetailsComponent,
    ParameterProductAddComponent
  ]
})
export class ParametersModule {}
