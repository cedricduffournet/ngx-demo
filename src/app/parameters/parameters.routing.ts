import { RouterModule, Routes } from '@angular/router';
import { ListParametersComponent } from '@app/parameters/containers';
import { ParameterCivilityComponent } from '@app/parameters/civility';
import { ParameterProductCategoryComponent } from '@app/parameters/product-category';
import { AuthGuard } from '@app/authentication/services';

export const routes: Routes = [
  {
    pathMatch: 'full',
    path: '',
    component: ListParametersComponent,
    data: { title: 'PARAMETERS' }
  },
  {
    path: 'civilities',
    component: ParameterCivilityComponent,
    canActivate: [AuthGuard],
    data: {
      roles: [
        'ROLE_CIVILITY_CREATE',
        'ROLE_CIVILITY_EDIT',
        'ROLE_CIVILITY_DELETE'
      ],
      title: 'CIVILITIES',
      previousLink: ['parameters']
    }
  },
  {
    path: 'productcategories',
    component: ParameterProductCategoryComponent,
    canActivate: [AuthGuard],
    data: {
      roles: [
        'ROLE_PRODUCT_CATEGORY_CREATE',
        'ROLE_PRODUCT_CATEGORY_EDIT',
        'ROLE_PRODUCT_CATEGORY_DELETE'
      ],
      title: 'PRODUCT_CATEGORIES',
      previousLink: ['parameters']
    }
  }
];
export const routing = RouterModule.forChild(routes);
