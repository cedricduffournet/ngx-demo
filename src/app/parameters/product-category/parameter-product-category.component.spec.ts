import { Component } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ParameterProductCategoryComponent } from '@app/parameters/product-category';

@Component({ selector: 'app-product-category-list-view', template: '' })
class ListProductCategoryViewStubComponent {}

describe('ParameterCivilityComponent', () => {
  let fixture: ComponentFixture<ParameterProductCategoryComponent>;
  let component: ParameterProductCategoryComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [ParameterProductCategoryComponent, ListProductCategoryViewStubComponent]
    });

    fixture = TestBed.createComponent(ParameterProductCategoryComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
