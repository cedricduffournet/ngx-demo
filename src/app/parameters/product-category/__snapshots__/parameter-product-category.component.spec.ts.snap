// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParameterProductCategoryComponent should be created 1`] = `
<app-parameter-product-category>
  <app-product-category-list-view />
</app-parameter-product-category>
`;
